### PR TITLE
Todo list Features

### PR DESCRIPTION
**Pull request by binhussen**
## Todo List Features
In this PR, I have implemented
- Added todo function (add, edit and delete todos) with form and handling event
- Apply style and react-icons on the app
- Persist data in local storage
- Used hooks and lifecycle methods
- Added Routing
- Added hamburger menu

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
+  render(<App />);
+  const linkElement = screen.getByText(/learn react/i);
+  expect(linkElement).toBeInTheDocument();
+});
```
Comment (Farida-Hamid) [inline src/App.test.js:8]: Great job building tests even though they are not required.

File: src/components/TodoContainer.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                    todos={todos}
+                    handleClick={this.handleClick}
+                    handleDelete={this.deleteItem}
+                    setUpdate={this.setUpdate}
+                  />
+                </div>
+              </div>
+          )}
+          />
+          <Route path="/about" element={<About />} />
```
Comment (Farida-Hamid) [inline src/components/TodoContainer.js:117]: Great job working with Routers, again, even though it's further in the curriculum.

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  <React.StrictMode>
+    <BrowserRouter basename={process.env.PUBLIC_URL}>
+      <App />
+    </BrowserRouter>
+  </React.StrictMode>,
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
```
Comment (Farida-Hamid) [inline src/index.js:19]: **OPTIONAL:** You may remove the comments since you got the message from the React team.

File: src/pages/About.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      slug: 'about-app',
+      title: 'About the App',
+      description: 'In this app, you can add, delete, submit and edit items. To edit items, simply double click on it. Once you are done, press the enter key to resubmit. This app will persist your data in the browser local storage. So whether you reload, close your app or reopened it, you still have access to your to-dos items.',
+    },
+    {
+      slug: 'about-author',
+      title: 'About the Author',
+      description: 'This app was developed by Ibas Majid, a self taught web developer and a technical writer. He is opened to freelance Gig. So go ahead and connect with ibas on Twitter @ibaslogic.',
+    },
+  ];
```
Comment (Farida-Hamid) [inline src/pages/About.js:17]: Neat and DRY 👍

**Review by Farida-Hamid — APPROVED** [review]
## Congratulations on building your web app.
You made an excellent project out of a small exercise. Your commitment and diligence are inspiring.

- ✅ Linters are checked.
- ✅ Best practices of JavaScript, HTML, and CSS are followed.
- ✅ The code follows DRY, YAGNI, and KISS.
- ✅ The code follows Gitflow.

 Keep it up.

File: src/index.js (hunk 3 above)
Comment (binhussen) [reply src/index.js:19]: Thanks, I will remove them.

File: src/pages/About.js (hunk 4 above)
Comment (binhussen) [reply src/pages/About.js:17]: kindly, It is sample JSON data that has to be rendered in the about section.

**Comment by binhussen** [comment]
Thank you @Farida-Hamid for your feedback, I will modify it.

**Pull request merged**